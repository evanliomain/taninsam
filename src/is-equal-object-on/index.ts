export * from './is-equal-object-on';
