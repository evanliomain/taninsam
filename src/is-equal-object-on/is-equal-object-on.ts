/**
 * @module object=>boolean
 */
/**
 * Compare 2 objects `o1` and `o2` on a set of property `mapping`
 * to determine if they are equals
 * @param iteratee The iteratee invoked per element.
 * @param mapping a list of key to compare `o1` and `o2`
 * @param o1 first object to compare
 * @param o2 second object to compare
 * @returns true if `o1[key] === o2[key]` for all key in mapping,
 * false otherwise
 * @example
 * ```
 * isEqualObjectOn(['a', 'b'])({ a: 1, b: 2, c: 3 })({ a: 1, b: 2, c: 4 })
 * // true
 * isEqualObjectOn(['a', 'b', 'c'])({ a: 1, b: 2, c: 3 })({ a: 1, b: 2, c: 4 })
 * // false
 * ```
 * @example Using the chain
 * ```
 * chain({ a: 1, b: 2, c: 4 })
 *   .chain(isEqualObjectOn(['a', 'b'])({ a: 1, b: 2, c: 3 }))
 *   .value() // true
 * ```
 */
export function isEqualObjectOn(
  mapping: string[]
): (o1: object) => (o1: object) => boolean {
  return (o1: object) => (o2: object) =>
    mapping.every(fieldName => o1[fieldName] === o2[fieldName]);
}
