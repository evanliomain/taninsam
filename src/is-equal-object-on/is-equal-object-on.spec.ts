import { isEqualObjectOn } from './is-equal-object-on';

describe('isEqualObjectOn function', () => {
  const input1 = { a: 1, b: 2, c: 3 };
  const input2 = { a: 1, b: 2, c: 4 };
  const mapping = ['a', 'b'];

  test('is a pure function', () => {
    isEqualObjectOn(mapping)(input1)(input2);
    expect(input1).toMatchSnapshot();
    expect(input2).toMatchSnapshot();
  });

  test('{ a: 1, b: 2, c: 4 } |> isEqualObjectOn(["a", "b"])({ a: 1, b: 2, c: 3 }) === true', () => {
    expect(isEqualObjectOn(mapping)(input1)(input2)).toBeTruthy();
  });

  test('{ a: 1, b: 2, c: 4 } |> isEqualObjectOn(["a", "b", "c"])({ a: 1, b: 2, c: 3 }) === true', () => {
    expect(isEqualObjectOn(['a', 'b', 'c'])(input1)(input2)).toBeFalsy();
  });
});
