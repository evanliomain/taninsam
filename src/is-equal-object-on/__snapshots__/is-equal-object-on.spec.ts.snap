// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`isEqualObjectOn function > is a pure function 1`] = `
{
  "a": 1,
  "b": 2,
  "c": 3,
}
`;

exports[`isEqualObjectOn function > is a pure function 2`] = `
{
  "a": 1,
  "b": 2,
  "c": 4,
}
`;
